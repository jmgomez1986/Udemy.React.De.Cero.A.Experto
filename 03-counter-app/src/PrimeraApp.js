// import React, { Fragment } from 'react';
import React from 'react';
import PropTypes from 'prop-types'

const PrimeraApp = ( {saludo, subtitulo} ) => {

	// const saludo = 'Hola Mundo!!!';
	// const objeto = {
	// 	nombre: 'Matias',
	// 	edad: 34,
	// };

	return (
		<>
			{/* <h1> { saludo } </h1> */}
			<h1>{ saludo }</h1>
			{/* <pre> { JSON.stringify(objeto, null, 3) } </pre> */}
			<p>{ subtitulo }</p>
		</>
	);
};

// Con los propTypes obligamos a que cuando se use el componente se envie la propiedad
PrimeraApp.propTypes = {
	saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
	subtitulo: 'Esto es un subt√≠tulo!!!'
}

export default PrimeraApp;
