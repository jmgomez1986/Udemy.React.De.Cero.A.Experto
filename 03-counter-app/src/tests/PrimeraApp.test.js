import React from 'react';
import '@testing-library/jest-dom';
// import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import PrimeraApp from '../PrimeraApp';

describe('Prueba en <PrimeraApp />', () => {
	// test('Debe mostrar el mensaje "Hola Matias" ', () => {
	// 	const saludo = 'Hola Matias';

	// 	const {getByText} = render(<PrimeraApp saludo={saludo} />);

	// 	expect(getByText(saludo)).toBeInTheDocument();

	// });

	test('debe de mostrar <PrimeraApp /> correctamente ', () => {
		const saludo = 'Hola Matias';
		const wrapper = shallow(<PrimeraApp saludo={saludo} />);

		expect(wrapper).toMatchSnapshot();
	});

	test('debe mostrar el subtitulo enviado por props ', () => {
		const saludo = 'Hola Matias';
		const sub = 'Subtitulo TEST';
		const wrapper = shallow(
			<PrimeraApp
				saludo={saludo}
				subtitulo={sub} />
		);
		const textoParrafo = wrapper.find('p').text();

		expect(textoParrafo).toBe(sub);
	});
});
