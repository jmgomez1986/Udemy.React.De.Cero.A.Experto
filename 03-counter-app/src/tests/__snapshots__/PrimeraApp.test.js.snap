// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <PrimeraApp /> debe de mostrar <PrimeraApp /> correctamente  1`] = `
<Fragment>
  <h1>
    Hola Matias
  </h1>
  <p>
    Esto es un subt√≠tulo!!!
  </p>
</Fragment>
`;
